### PR TITLE
GP2-3283 Delete export plan

### PR DESCRIPTION
**Pull request by bobmeredith**
Delete export plan.
Adds a button on the EP homepage to delete just that one plan.
There is a confirmation modal when the delete button is clicked to prevent accidental deletion.  Because of this, it's a largely React implementation, calling an Ajax endpoint to do the deletion and then jumping to the EP index page if successful.

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/GP2-3283
- [x] Jira ticket has the correct status.
- [x] [Changelog](CHANGELOG.md) entry added.

### Reviewing help

- [ ] Explains how to test locally, including how to set up appropriate data
- [x] Includes screenshot(s) - ideally before and after, but at least after

### Housekeeping

- [ ] Added all new environment variables to Vault.
- [ ] Cleaned up old feature flags
- [ ] Upgraded any vulnerable dependencies.
- [ ] Ran the `make manage download_geolocation_data` command
- [ ] I have updated security dependencies
- [x] Python requirements have been re-compiled.
- [x] Frontend assets have been re-compiled.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

**Comment by bobmeredith** [comment]
image

image

File: exportplan/templates/exportplan/dashboard_page.html [hunk 1]
```diff
@@ -67,10 +67,12 @@ <h3 class="m-v-xs bold">Upload your logo</h3>
     </div>
     {% endif %}
     {% with export_plan_id=export_plan.pk %}
-    <a class="button button--secondary button--full-width button--icon m-b-xs" href="{{export_plan_download_link}}" title="Save your plan as a PDF">
+    <a class="button button--secondary button--small button--full-width button--icon m-b-xs" href="{{export_plan_download_link}}" title="Save your plan as a PDF">
```
Comment (froddd) [inline exportplan/templates/exportplan/dashboard_page.html:70]: `title` attribute needs updating to new wording

File: exportplan/core/helpers.py [hunk 2]
```diff
@@ -13,6 +13,12 @@ def create_export_plan(sso_session_id, data):
     return response.json()
 
 
+def delete_export_plan(sso_session_id, id):
+    response = api_client.exportplan.delete_export_plan(sso_session_id=sso_session_id, id=id)
```
Comment (froddd) [inline exportplan/core/helpers.py:17]: I take it the `delete_export_plan` methods makes sure the export plan ID belongs to the session user? Just checking ;)

File: react-components/src/components/Dashboard/DeleteButton.test.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      const { queryByText, getByText } = setup()
+      const button = getByText('Delete plan')
+      expect(button).toBeInTheDocument()
+      fireEvent.click(button)
+      expect(getByText(confirmationTitle)).toBeInTheDocument()
+      fireEvent.click(getByText('Cancel'))
+      expect(queryByText(confirmationTitle)).toBeFalsy()
+    })
+
+    xit('Should delete export plan', () => {
```
Comment (froddd) [inline react-components/src/components/Dashboard/DeleteButton.test.jsx:38]: Should this be skipped?

File: react-components/src/views/sections/Dashboard/index.jsx [hunk 4]
```diff
@@ -1,8 +1,19 @@
...
 import React from 'react'
+import { connect, Provider } from 'react-redux'
 
+import Services from '@src/Services'
 import { Dashboard } from '@src/components/Dashboard'
+import DeleteButton from '@src/components/Dashboard/DeleteButton'
 
 export const createDashboard = ({ element, ...params }) => {
   ReactDOM.render(<Dashboard {...params} />, element)
+  const deleteButton = document.body.querySelector('.export-plan-delete')
```
Comment (froddd) [inline react-components/src/views/sections/Dashboard/index.jsx:11]: If there's only going to be one deletable plan on any page, would it make sense to use `id` for consistency?

File: react-components/src/components/Dashboard/DeleteButton.test.jsx (hunk 3 above)
Comment (bobmeredith) [reply react-components/src/components/Dashboard/DeleteButton.test.jsx:38]: Nice spot - No, I was fixing for the last name change and missed turning back on.

File: react-components/src/views/sections/Dashboard/index.jsx (hunk 4 above)
Comment (bobmeredith) [reply react-components/src/views/sections/Dashboard/index.jsx:11]: Good point. Will change.
Annoyingly, it seems that the designers want an EP delete from the list page too which means this whole mechanism needs to change to get any re-use.

File: exportplan/core/helpers.py (hunk 2 above)
Comment (bobmeredith) [reply exportplan/core/helpers.py:17]: Actually no, but then none of the endpoints do.  I created a ticket to fix this as a global decorator probably in d-api. It's with @depsiatwal for this sprint.

File: exportplan/templates/exportplan/dashboard_page.html (hunk 1 above)
Comment (bobmeredith) [reply exportplan/templates/exportplan/dashboard_page.html:70]: Oh yes - actually, need to check with content and see if they want more detail there. Not much point in a title that matches the button label I guess.

Comment (bobmeredith) [reply exportplan/templates/exportplan/dashboard_page.html]: Yeah - got new titles from content 'Delete/Download your export plan'

**Pull request merged**